src/Compilers/CSharp/Test/Syntax/Parsing/ParsingErrorRecoveryTests.cs

```diff
@@ -5034,9 +5034,8 @@ public void TestSemicolonAfterObjectInitializerMemberComma()
             Assert.NotEqual(SyntaxKind.None, ds.Declaration.Variables[0].Initializer.EqualsToken.Kind());
             Assert.NotNull(ds.Declaration.Variables[0].Initializer.Value);
             Assert.Equal(SyntaxKind.ObjectCreationExpression, ds.Declaration.Variables[0].Initializer.Value.Kind());
-            Assert.Equal(2, file.Errors().Length);
-            Assert.Equal((int)ErrorCode.ERR_InvalidExprTerm, file.Errors()[0].Code);
-            Assert.Equal((int)ErrorCode.ERR_RbraceExpected, file.Errors()[1].Code);
+            Assert.Equal(1, file.Errors().Length);
+            Assert.Equal((int)ErrorCode.ERR_RbraceExpected, file.Errors()[0].Code);
```

Comment:
moving to the new helper produced better errors for object/collection initializer parsing.  The old logic would always attempt to parse out an element after a comma.  The new logic (shared with the rest of separated list parsing) only does that if what follows the comma could start an element.  This prevents a dual error of not having the write element type *and* then saying you need to close the teh initializer.  Now it just says that the initializer needs to be closed (like all other list parsing).